pkgs/by-name/te/teams-for-linux/package.nix

```
@@ -0,0 +1,128 @@
+    rev = "v${version}";
+    hash = "sha256-+rEGDg+/qvjCMhGHccb4p+CKOo/65RpkFT/WnCDlCgU=";
+  };
+
+  npmDepsHash = "sha256-vDRFFxkIQo5qU9gmkSwUhPz4FG2XbUNkTw6SCuvMqCc=";
+
+  nativeBuildInputs =
+    [
+      makeWrapper
+      nodejs
+    ]
+    ++ lib.optionals (stdenvNoCC.isLinux) [
+      copyDesktopItems
+    ];
+
+  env = {
+    # disable code signing on Darwin
+    CSC_IDENTITY_AUTO_DISCOVERY = lib.optionalString stdenvNoCC.isDarwin "false";
+    ELECTRON_SKIP_BINARY_DOWNLOAD = "1";
+  };
+
+  buildPhase = ''
+    cp -r ${electron_30.dist} electron-dist
+    chmod -R u+w electron-dist
+
+    npm exec electron-builder -- \
+        --dir \
+        -c.npmRebuild=true \
+        -c.asarUnpack="**/*.node" \
+        -c.electronDist=electron-dist \
+        -c.electronVersion=${electron_30.version}
+  '';
+
+  installPhase =
+    ''
+      runHook preInstall
+
+    ''
+    + lib.optionalString stdenvNoCC.isLinux ''
+      mkdir -p $out/share/{applications,teams-for-linux}
+      cp dist/*-unpacked/resources/app.asar $out/share/teams-for-linux/
+
+      pushd build/icons
+      for image in *png; do
+        mkdir -p $out/share/icons/hicolor/''${image%.png}/apps
+        cp -r $image $out/share/icons/hicolor/''${image%.png}/apps/teams-for-linux.png
+      done
+      popd
+
+      # Linux needs 'aplay' for notification sounds
+      makeWrapper '${electron_30}/bin/electron' "$out/bin/teams-for-linux" \
+        --prefix PATH : ${
+          lib.makeBinPath [
+            alsa-utils
+            which
+          ]
+        } \
+        --add-flags "$out/share/teams-for-linux/app.asar" \
+        --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}"
+    ''
+    + lib.optionalString stdenvNoCC.isDarwin ''
+      mkdir -p $out/Applications
+      cp -r dist/mac*/teams-for-linux.app $out/Applications
+      makeWrapper $out/Applications/teams-for-linux.app/Contents/MacOS/teams-for-linux $out/bin/teams-for-linux
+    ''
+    + ''
+
+      runHook postInstall
+    '';
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "teams-for-linux";
+      exec = "teams-for-linux";
+      icon = "teams-for-linux";
+      desktopName = "Microsoft Teams for Linux";
+      comment = meta.description;
+      categories = [
+        "Network"
+        "InstantMessaging"
+        "Chat"
+      ];
+    })
+  ];
+
+  passthru.updateScript = nix-update-script { };
+  passthru.tests.version = testers.testVersion {
+    package = teams-for-linux;
+    command = "HOME=$TMPDIR teams-for-linux --version";
+  };
+
+  meta = {
+    description = "Unofficial Microsoft Teams client for Linux";
+    mainProgram = "teams-for-linux";
+    homepage = "https://github.com/IsmaelMartinez/teams-for-linux";
+    license = lib.licenses.gpl3Only;
+    maintainers = with lib.maintainers; [
+      muscaln
+      qjoly
+      chvp
+      khaneliman
+    ];
+    inherit (electron_30.meta) platforms;
```

```suggestion
    platforms = with lib.platforms; darwin ++ linux;
```
because `installPhase` doesn't do anything elsewhere.

---

Makes sense, will update.